### PR TITLE
Improve nav menu filtering and allow navigating using keyboard (arrows and enter keys)

### PR DESCRIPTION
**Pull request by xantorres**
Missing: keep section expanded after navigating to a sublink, improve styling

navbar

### CODE REVIEW

File: src/app/core/common/HotKeysProvider.js [hunk 1]
```diff
@@ -48,19 +49,18 @@ export default class HotKeysProvider extends Component {
...
   }
 
   handleKeyDown = e => {
     const hotkeyHandlers = this.state.hotKeyHandlers[e.key]
-
     if (hotkeyHandlers &&
       !isEmpty(hotkeyHandlers) &&
       !e.metaKey && !e.altKey && !e.shiftKey &&
-      !this.isEditable(e)) {
+      (editEndKeys.includes(e.key) || !this.isEditable(e))) {
```
Comment (brennancheung) [inline src/app/core/common/HotKeysProvider.js]: This is getting messy.  What about moving all the boolean conditionals into their own function?

File: src/app/core/common/Navbar.js [hunk 2]
```diff
@@ -124,12 +134,48 @@ class Navbar extends PureComponent {
...
   }
 
+  handleArrowKeys = direction => () => {
+    const { filterText, activeNavItem } = this.state
+    if (filterText && activeNavItem) {
+      // Highlight next nav item
+      const offset = direction === 'ArrowDown' ? 1 : -1
+      const sectionLinks = this.getSectionLinks()
+      const currentIdx = sectionLinks.findIndex(propEq('name', activeNavItem))
+      const nextIdx = currentIdx + offset >= sectionLinks.length ? 0
```
Comment (brennancheung) [inline src/app/core/common/Navbar.js]: Getting a bit complicated here.  Is this just for wrapping around?  What about using modulus?

`(currentIdx + offset) % sectionLinks.length`

File: src/app/core/common/Navbar.js [hunk 3]
```diff
@@ -246,35 +321,34 @@ class Navbar extends PureComponent {
...
+            <ExpansionPanelDetails className={classes.navBody}>
+              <MenuList component="nav" className={classes.navMenu}>
+                {(filterText
+                  ? this.getFilteredLinks(section.links)
+                  : section.links
+                ).map(this.renderNavLink)}
+              </MenuList>
+            </ExpansionPanelDetails>
+          </ExpansionPanel>)
+        : <MenuList component="nav" className={classes.navMenu}>
```
Comment (brennancheung) [inline src/app/core/common/Navbar.js]: There's too many lines between the ternary operator terms.  What about just making it more explicit?

```javascript
{filteredSections.length > 1 && ...}
{filteredSections.length <= 1 && ..}
```

Comment (brennancheung) [reply src/app/core/common/Navbar.js]: Moving the blocks into functions might work as well and even be cleaner.

Comment (xantorres) [reply src/app/core/common/Navbar.js]: Yeah I had pending doing a little refactoring and cleanup

File: src/app/core/common/Navbar.js (hunk 2 above)
Comment (xantorres) [reply src/app/core/common/Navbar.js]: I want to go to iterate between first/last items when reaching the start/end of the list, that's why I do this. 
I will try to simplify this a little more or at least add a comment explain the goal of this.

Comment (xantorres) [reply src/app/core/common/Navbar.js]: Good advice, that worked!

**Pull request merged**